react/features/toolbar/components/Toolbar.web.js

```diff
@@ -0,0 +1,409 @@
+/* @flow */
+
+declare var AJS: Object;
+declare var APP: Object;
+declare var config: Object;
+declare var interfaceConfig: Object;
+declare var $: Function;
+
+import { connect } from 'react-redux';
+import React from 'react';
+
+import { FeedbackButton } from '../../feedback';
+import Recording from '../../../../modules/UI/recording/Recording';
+import UIUtil from '../../../../modules/UI/util/UIUtil';
+import UIEvents from '../../../../service/UI/UIEvents';
+
+import { AbstractToolbar, _mapStateToProps } from './AbstractToolbar';
+import DEFAULT_TOOLBAR_BUTTONS from './defaultToolbarButtons';
+import primaryToolbarHandlers from './primaryToolbarHandlers';
+import secondaryToolbarHandlers from './secondaryToolbarHandlers';
+import ToolbarButton from './ToolbarButton';
+
+/**
+ * Handlers for toolbar buttons.
+ *
+ * buttonId {string}: handler {function}
+ */
+const BUTTON_HANDLERS = Object.assign({}, primaryToolbarHandlers,
+    secondaryToolbarHandlers);
+
+/**
+ * For legacy reasons, inline style for display none.
+ *
+ * @private
+ * @type {{
+ *     display: string
+ * }}
+ */
+const _DISPLAY_NONE_STYLE = {
+    display: 'none'
+};
+
+/**
+ * Implements the conference toolbar on React.
+ *
+ * @extends AbstractToolbar
+ */
+class Toolbar extends AbstractToolbar {
+
+    /**
+     * Constructor of Toolbar component.
+     *
+     * @param {Object} props - The read-only React Component props with which
+     * the new instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        // Bind methods to save the context
+        const self: any = this;
+
+        self._onLocalRaiseHandChanged
+            = this._onLocalRaiseHandChanged.bind(this);
+        self._onFullScreenToggled = this._onFullScreenToggled.bind(this);
+    }
+
+    /**
+     * Toolbar component's property types.
+     *
+     * @static
+     */
+    static propTypes = AbstractToolbar.propTypes;
+
+    /**
+     * Defines buttons to render in primary and secondary toolbars.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillMount() {
+        const {
+            primaryToolbar,
+            secondaryToolbar
+        } = interfaceConfig.TOOLBAR_BUTTONS.reduce((acc, buttonName) => {
+            const button = DEFAULT_TOOLBAR_BUTTONS[buttonName];
+
+            if (button) {
+                const place = this._getToolbarButtonPlace(buttonName);
+
+                button.buttonName = buttonName;
+                acc[place].set(button.id, button);
+            }
+
+            return acc;
+        }, {
+            primaryToolbar: new Map(),
```

Comment:
Arrays suits here but in some cases we need to get elements by key. If we represent data via arrays it costs O(n) to access the element. Maps allow us to get cheap element access (O(1) mostly) and we can traverse them in loop (because they are ordered) just like arrays.